src/screens/NationDetailsContainer/index.js

```diff
@@ -1,22 +1,73 @@
+// @flow
+
 import React from 'react';
-import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
+import { Alert } from 'react-native';
+import _ from 'lodash';
 
 import NationDetailsScreen from './NationDetailsScreen';
 import { switchNationTab, joinNation, leaveNation } from '../../actions/nations';
 import { androidNavigationButtons, screen } from '../../global/Screens';
-import { Alert } from 'react-native';
 import i18n from '../../global/i18n';
 import Colors from '../../global/colors';
-import { deleteNationDraft, startNationEditing, submitNation } from '../../actions/modifyNation';
-import { openedNation } from '../../reducers/nations';
+import {
+  deleteNationDraft,
+  startNationEditing,
+  submitNation,
+} from '../../actions/modifyNation';
+import { openedNation, type State as NationState } from '../../reducers/nations';
 import NavigatorComponent from '../../components/common/NavigatorComponent';
 import { alert, errorAlert } from '../../global/alerts';
 import { nationIsDraft } from '../../utils/nations';
+import type { Navigator } from '../../types/ReactNativeNavigation';
+import type { NationIdType, NationType } from '../../types/Nation';
 
 const EDIT_BUTTON = 'EDIT_BUTTON';
 
-class NationDetailsContainer extends NavigatorComponent {
+type Props = {
+  /**
+   * @desc React Native Navigation navigator object.
+   */
+  navigator: Navigator,
+  /**
+   * @desc A boolean prop to indicate whether the nation is Draft or not
+   */
+  isDraft: boolean,
+}
+
+type Actions = {
+  /**
+   * @desc Function to select the tab on screen
+   * @param index Id of the tab to appear on screen
+   */
+  onSelectTab: (number) => void,
+  /**
+   * @desc Function to join a nation
+   */
+  joinNation: () => void,
+  /**
+   * @desc Function to leave a nation
+   */
+  leaveNation: () => void,
+  /**
+   * @desc Function to start the edition of a nation draft
+   * @param nation The empty nation's draft
+   */
+  onStartNationEditing: (NationType) => void,
+  /**
+   * @desc Function to delete a nation draft
+   * @param nationId The Id of the nation to delete
+   */
+  onDeleteDraft: (NationIdType, () => void) => void,
```

Comment:
Please add description of second parameter as well.

Reply:
Done